### PR TITLE
Elasticsearch, Pycsw and setup.py

### PR DESCRIPTION
**Pull request by ingenieroariel**
This branch is the work in progress (with @jj0hns0n and @tomkralidis) to enable pycsw and elasticsearch on HHyperMap. I am creating it as a placeholder and will make sure it applies cleanly on master.

This work was commissioned by BoundlessGeo.

@capooti Let's try to get this merged before next week to make sure all future contributions can be merged back upstream and we don't diverge too much.
 - [x] Move cf folder to deploy and move ansible in there too. (Jorge)
 - [x] Get rid of _vcap.settings (Jorge)
 - [x] MAke sure latest pycsw work is merged in this branch and that we are pointing to the latest alpha on pypi. (Tom)
 - [x] Make sure Travis passes (Ariel)
 - [x] Squash it (Ariel)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -69,3 +69,8 @@ ec2.py
 
 hypermap/media
 hypermap/settings/local_settings.py
+
+# IDE specific
+/hypermap/settings/_panchicore.py
```
Comment (capooti) [inline .gitignore]: what is it? (just curious :) )

File: hypermap/hypermap/urls.py [hunk 2]
```diff
@@ -11,14 +12,14 @@
...
 )
 
-if settings.DEBUG:
-    urlpatterns += patterns(
-        '',
-        url(r'^media/(?P<path>.*)$',
-            'django.views.static.serve',
-            {'document_root': settings.MEDIA_ROOT, }),
-    )
+urlpatterns += patterns(
```
Comment (capooti) [inline hypermap/hypermap/urls.py]: shouldn't this be a nginx task in production?

File: hypermap/settings/_vcap.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import dj_database_url
+import json
+import os
+from settings.default import *
+
+vcap_service_config = os.environ.get('VCAP_SERVICES')
```
Comment (capooti) [inline hypermap/settings/_vcap.py]: what vcap stands for?

File: hypermap/settings/_vcap.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+vcap_app_config = os.environ.get('VCAP_APPLICATION')
+#TODO: Get from environment and don't hardcode
+SITE_URL = 'hypermap-demo.cfapps.io'
+
+ALLOWED_HOSTS = [SITE_URL, 'localhost']
+
+DATABASES = {'default': dj_database_url.config()}
+
+BROKER_DB = 0
+BROKER_URL = 'redis://:{0}@{1}:{2}/{3}'.format(
```
Comment (capooti) [inline hypermap/settings/_vcap.py]: so you are using redis in place of rabbit? any reason?

File: hypermap/hypermap/urls.py (hunk 2 above)
Comment (ingenieroariel) [inline hypermap/hypermap/urls.py]: We are deploying this in an environment that does not let us configure nginx.

In your case, nginx will serve this request before it gets to django, so it is  non issue.

File: hypermap/settings/_vcap.py (hunk 3 above)
Comment (ingenieroariel) [inline hypermap/settings/_vcap.py]: This is the deployment environment of cloud foundry - where we are deploying hypermap.

File: hypermap/settings/_vcap.py (hunk 4 above)
Comment (ingenieroariel) [inline hypermap/settings/_vcap.py]: We were trying it out - but will go back to Rabbit soon

File: .gitignore (hunk 1 above)
Comment (ingenieroariel) [inline .gitignore]: @panchicore is a crazy colombian developer who is working on scaling. I don't think we need this change in the long run.

**Pull request merged**